src/nvim/channel.c

```diff
@@ -834,3 +836,55 @@ Array channel_all_info(void)
   });
   return ret;
 }
+
+void invoke_prompt_callback(void)
```

Comment:
This doesn't seem to have anything to do with (nvim) channels. Perhaps it should be in `eval.c`